Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/PopupMenuController.java

```diff
@@ -0,0 +1,233 @@
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+
+package org.apromore.portal.dialogController;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.apromore.plugin.portal.PortalContext;
+import org.apromore.plugin.portal.PortalLoggerFactory;
+import org.apromore.plugin.portal.PortalPlugin;
+import org.apromore.portal.common.LabelConstants;
+import org.apromore.portal.context.PortalPluginResolver;
+import org.apromore.portal.menu.MenuConfig;
+import org.apromore.portal.menu.MenuConfigLoader;
+import org.apromore.portal.menu.MenuGroup;
+import org.apromore.portal.menu.MenuItem;
+import org.apromore.portal.menu.PluginCatalog;
+import org.apromore.portal.model.UserType;
+import org.slf4j.Logger;
+import org.zkoss.spring.SpringUtil;
+import org.zkoss.util.resource.Labels;
+import org.zkoss.zk.ui.Executions;
+import org.zkoss.zk.ui.Sessions;
+import org.zkoss.zk.ui.select.SelectorComposer;
+import org.zkoss.zk.ui.select.annotation.VariableResolver;
+import org.zkoss.zul.Menuitem;
+import org.zkoss.zul.Menupopup;
+import org.zkoss.zul.Menuseparator;
+import org.zkoss.zul.Messagebox;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.google.common.base.Strings;
+
+@VariableResolver(org.zkoss.zkplus.spring.DelegatingVariableResolver.class)
+public class PopupMenuController extends SelectorComposer<Menupopup> {
+	private static final Logger LOGGER = PortalLoggerFactory.getLogger(BaseMenuController.class);
+
+	protected transient MenuConfigLoader menuConfigLoader;
+	protected transient Map<String, PortalPlugin> portalPluginMap;
+	private static final String DISPLAY_NAME_EXP = "displayName";
+	private static final String GROUP = "group";
+	private static final String ON_CLICK = "onClick";
+
+	@Override
+	public void doAfterCompose(Menupopup menuPopup) {
+		try {
+			String popupTYpe = (String) Executions.getCurrent().getArg().get("POPUP_TYPE");
+			if (!PortalPluginResolver.resolve().isEmpty()) {
+				if ("FOLDER".equals(popupTYpe)) {
+					loadPopupMenu(menuPopup, "folder-popup-menu");
+				} else if ("PROCESS".equals(popupTYpe)) {
+					loadPopupMenu(menuPopup, "process-popup-menu");
+				} else if ("LOG".equals(popupTYpe)) {
+					loadPopupMenu(menuPopup, "log-popup-menu");
+				}
+			}
```

Comment:
Typo: `popupTYpe` should be `popupType'.

The chain of conditionals might work more succinctly as a switch.

Reply:
Rewrote with the switch myself.